### PR TITLE
[fix]: [PL-25096]: Improving the Authentication APIs Summary and desc…

### PR DESCRIPTION
**Pull request by boopesh007**
…ription wrt OpenApiSpec

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/34074)

### CODE REVIEW

**Comment by boopesh007** [comment]
trigger messagecheck

File: 400-rest/src/main/java/software/wings/resources/UserGroupResource.java [hunk 1]
```diff
@@ -406,11 +417,19 @@ public RestResponse<UserGroup> unlinkSsoGroup(@PathParam("userGroupId") String u
    */
   @PUT
   @Path("{userGroupId}/link/saml/{samlId}")
+  @ApiOperation(value = "Link Harness User Group with SAML group", nickname = "linkUserGroupWithSAMLGroup")
+  @Operation(operationId = "linkUserGroupWithSAMLGroup", summary = "Link a User Group to a SAML Group.",
```
Comment (ashishsanodia-harness) [inline 400-rest/src/main/java/software/wings/resources/UserGroupResource.java:421]: Remove full stop from summary

**Comment by boopesh007** [comment]
*[`400-rest/src/main/java/software/wings/resources/UserGroupResource.java` line 421 at r3](https://reviewable.io/reviews/harness/harness-core/34074#-N3zShItvmPc_VxwrIlf-r3-421:-N4Bkn42822l-LiG4aQB:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/781634096844344d05dbf9c0d5252ec44d8d87db/400-rest/src/main/java/software/wings/resources/UserGroupResource.java#L421)):*
<details><summary><i>Previously, ashishsanodia-harness (Ashish Sanodia) wrote…</i></summary><blockquote>

Remove full stop from summary
</blockquote></details>

Done.

**Review by ashishsanodia-harness — DISMISSED** [review]
> Reviewed 1 of 3 files at r1, 1 of 2 files at r2, 1 of 2 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34074)* status: 4 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

File: 120-ng-manager/src/main/java/io/harness/ng/scim/resource/NGScimUserResource.java [hunk 2]
```diff
@@ -128,6 +132,7 @@ public class NGScimUserResource extends ScimResource {
 
   @GET
   @Path("Users")
+  @Hidden
   @ApiOperation(
       value =
           "Search users by their email address. Supports pagination. If nothing is passed in filter, all results will be returned.",
```
Comment (rashminandasahoo) [inline 120-ng-manager/src/main/java/io/harness/ng/scim/resource/NGScimUserResource.java:138]: Search for users by email addresses. If the filter parameter is left blank, all the users will be returned.

File: 120-ng-manager/src/main/java/io/harness/ng/scim/resource/NGScimUserResource.java [hunk 3]
```diff
@@ -156,6 +161,7 @@ public class NGScimUserResource extends ScimResource {
 
   @DELETE
   @Path("Users/{userIdentifier}")
+  @Hidden
   @ApiOperation(value = "Delete an user by uuid", nickname = "deleteScimUser")
   @Operation(operationId = "deleteScimUser", summary = "Deletes user by ID",
```
Comment (rashminandasahoo) [inline 120-ng-manager/src/main/java/io/harness/ng/scim/resource/NGScimUserResource.java:166]: Deletes user for the given ID.

File: 120-ng-manager/src/main/java/io/harness/ng/scim/resource/NGScimUserResource.java [hunk 4]
```diff
@@ -172,6 +178,7 @@ public class NGScimUserResource extends ScimResource {
 
   @PATCH
   @Path("Users/{userIdentifier}")
+  @Hidden
   @ApiOperation(value = "Update some fields of a user by uuid", nickname = "patchScimUser")
```
Comment (rashminandasahoo) [inline 120-ng-manager/src/main/java/io/harness/ng/scim/resource/NGScimUserResource.java:182]: Updates user for the given UUID.

File: 122-ng-authentication-settings/src/main/java/io/harness/ng/authenticationsettings/resources/AuthenticationSettingsResource.java [hunk 5]
```diff
@@ -163,11 +163,12 @@ public class AuthenticationSettingsResource {
   @PUT
   @Path("/oauth/update-providers")
   @ApiOperation(value = "Update Oauth providers for an account", nickname = "updateOauthProviders")
-  @Operation(operationId = "updateOauthProviders", summary = "Updates the Oauth providers by accountIdentifier",
+  @Operation(operationId = "updateOauthProviders", summary = "Update Oauth providers",
```
Comment (rashminandasahoo) [inline 122-ng-authentication-settings/src/main/java/io/harness/ng/authenticationsettings/resources/AuthenticationSettingsResource.java:166]: Updates OAuth providers for the given Account ID.

File: 122-ng-authentication-settings/src/main/java/io/harness/ng/authenticationsettings/resources/AuthenticationSettingsResource.java [hunk 6]
```diff
@@ -223,11 +226,12 @@ public class AuthenticationSettingsResource {
   @PUT
   @Path("/whitelisted-domains")
   @ApiOperation(value = "Update Whitelisted domains for an account", nickname = "updateWhitelistedDomains")
-  @Operation(operationId = "updateWhitelistedDomains", summary = "Updates the Whitelisted domains by accountIdentifier",
+  @Operation(operationId = "updateWhitelistedDomains", summary = "Update Whitelisted domains",
```
Comment (rashminandasahoo) [inline 122-ng-authentication-settings/src/main/java/io/harness/ng/authenticationsettings/resources/AuthenticationSettingsResource.java:229]: Updates the whitelisted domains.

File: 122-ng-authentication-settings/src/main/java/io/harness/ng/authenticationsettings/resources/AuthenticationSettingsResource.java [hunk 7]
```diff
@@ -286,11 +291,12 @@ public class AuthenticationSettingsResource {
   @Path("/saml-metadata-upload")
   @Consumes("multipart/form-data")
   @ApiOperation(value = "Edit SAML Config", nickname = "updateSamlMetaData")
-  @Operation(operationId = "updateSamlMetaData", summary = "Updates the SAML metadata by accountId",
+  @Operation(operationId = "updateSamlMetaData", summary = "Update SAML metadata",
```
Comment (rashminandasahoo) [inline 122-ng-authentication-settings/src/main/java/io/harness/ng/authenticationsettings/resources/AuthenticationSettingsResource.java:294]: Updates the SAML metadata for the given Account ID.

File: 122-ng-authentication-settings/src/main/java/io/harness/ng/authenticationsettings/resources/AuthenticationSettingsResource.java [hunk 8]
```diff
@@ -328,11 +334,12 @@ public class AuthenticationSettingsResource {
   @DELETE
   @Path("/delete-saml-metadata")
   @ApiOperation(value = "Delete SAML Config", nickname = "deleteSamlMetaData")
-  @Operation(operationId = "deleteSamlMetaData", summary = "Deletes SAML meta data by accountIdentifier",
+  @Operation(operationId = "deleteSamlMetaData", summary = "Delete SAML meta data",
```
Comment (rashminandasahoo) [inline 122-ng-authentication-settings/src/main/java/io/harness/ng/authenticationsettings/resources/AuthenticationSettingsResource.java:337]: Deletes SAML metadata for the given Account ID.

**Comment by boopesh007** [comment]
*[`120-ng-manager/src/main/java/io/harness/ng/scim/resource/NGScimUserResource.java` line 114 at r3](https://reviewable.io/reviews/harness/harness-core/34074#-N41F5CkYfjCQkKZgztO-r3-114:-N4C4Mkc4GQmO-P2eAX1:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/781634096844344d05dbf9c0d5252ec44d8d87db/120-ng-manager/src/main/java/io/harness/ng/scim/resource/NGScimUserResource.java#L114)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

Gets the user details for the given Account ID and User ID.
</blockquote></details>

Done.

**Comment by boopesh007** [comment]
*[`120-ng-manager/src/main/java/io/harness/ng/scim/resource/NGScimUserResource.java` line 94 at r3](https://reviewable.io/reviews/harness/harness-core/34074#-N41F5CkYfjCQkKZgztO-r3-94:-N4C4TS5C_UwkG_b9zYN:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/781634096844344d05dbf9c0d5252ec44d8d87db/120-ng-manager/src/main/java/io/harness/ng/scim/resource/NGScimUserResource.java#L94)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

Please change the casing of SCIM and UUID in all the places.
</blockquote></details>

Done.

**Comment by boopesh007** [comment]
*[`120-ng-manager/src/main/java/io/harness/ng/scim/resource/NGScimUserResource.java` line 166 at r3](https://reviewable.io/reviews/harness/harness-core/34074#-N41F5CkYfjCQkKZgztO-r3-166:-N4C5CX91vY24fjC3vuH:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/781634096844344d05dbf9c0d5252ec44d8d87db/120-ng-manager/src/main/java/io/harness/ng/scim/resource/NGScimUserResource.java#L166)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

Deletes user for the given ID.
</blockquote></details>

Done.

**Comment by boopesh007** [comment]
*[`120-ng-manager/src/main/java/io/harness/ng/scim/resource/NGScimUserResource.java` line 138 at r3](https://reviewable.io/reviews/harness/harness-core/34074#-N41F5CkYfjCQkKZgztO-r3-138:-N4C5-fdCr9Ybb8BRJef:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/781634096844344d05dbf9c0d5252ec44d8d87db/120-ng-manager/src/main/java/io/harness/ng/scim/resource/NGScimUserResource.java#L138)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

Search for users by email addresses. If the filter parameter is left blank, all the users will be returned.
</blockquote></details>

Done.

**Comment by boopesh007** [comment]
*[`120-ng-manager/src/main/java/io/harness/ng/scim/resource/NGScimUserResource.java` line 182 at r3](https://reviewable.io/reviews/harness/harness-core/34074#-N41F5CkYfjCQkKZgztO-r3-182:-N4C5dQv1JCpMIftwTh5:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/781634096844344d05dbf9c0d5252ec44d8d87db/120-ng-manager/src/main/java/io/harness/ng/scim/resource/NGScimUserResource.java#L182)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

Updates user for the given UUID.
</blockquote></details>

Done.

**Comment by boopesh007** [comment]
*[`122-ng-authentication-settings/src/main/java/io/harness/ng/authenticationsettings/resources/AuthenticationSettingsResource.java` line 101 at r3](https://reviewable.io/reviews/harness/harness-core/34074#-N3zShItvmPc_VxwrIle-r3-101:-N4C5nvf1wKjPBEwFEIA:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/781634096844344d05dbf9c0d5252ec44d8d87db/122-ng-authentication-settings/src/main/java/io/harness/ng/authenticationsettings/resources/AuthenticationSettingsResource.java#L101)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

Gets authentication settings for the given Account ID.
</blockquote></details>

Done.

**Comment by boopesh007** [comment]
*[`122-ng-authentication-settings/src/main/java/io/harness/ng/authenticationsettings/resources/AuthenticationSettingsResource.java` line 121 at r3](https://reviewable.io/reviews/harness/harness-core/34074#-N3zShItvmPc_VxwrIle-r3-121:-N4C5yKN4sHZcTJKYaUP:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/781634096844344d05dbf9c0d5252ec44d8d87db/122-ng-authentication-settings/src/main/java/io/harness/ng/authenticationsettings/resources/AuthenticationSettingsResource.java#L121)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

Gets password strength for the given Account ID.
</blockquote></details>

Done.

**Comment by boopesh007** [comment]
*[`122-ng-authentication-settings/src/main/java/io/harness/ng/authenticationsettings/resources/AuthenticationSettingsResource.java` line 166 at r3](https://reviewable.io/reviews/harness/harness-core/34074#-N3zShItvmPc_VxwrIle-r3-166:-N4C668f3DWymGVz9giY:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/781634096844344d05dbf9c0d5252ec44d8d87db/122-ng-authentication-settings/src/main/java/io/harness/ng/authenticationsettings/resources/AuthenticationSettingsResource.java#L166)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

Updates OAuth providers for the given Account ID.
</blockquote></details>

Done.

**Comment by boopesh007** [comment]
*[`122-ng-authentication-settings/src/main/java/io/harness/ng/authenticationsettings/resources/AuthenticationSettingsResource.java` line 189 at r3](https://reviewable.io/reviews/harness/harness-core/34074#-N3zShItvmPc_VxwrIle-r3-189:-N4C6EykCfRkmWqUYiNX:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/781634096844344d05dbf9c0d5252ec44d8d87db/122-ng-authentication-settings/src/main/java/io/harness/ng/authenticationsettings/resources/AuthenticationSettingsResource.java#L189)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

Deletes OAuth settings for a given Account ID.
</blockquote></details>

Done.

**Comment by boopesh007** [comment]
*[`122-ng-authentication-settings/src/main/java/io/harness/ng/authenticationsettings/resources/AuthenticationSettingsResource.java` line 208 at r3](https://reviewable.io/reviews/harness/harness-core/34074#-N3zShItvmPc_VxwrIle-r3-208:-N4C6NSN1xfohspOzz6u:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/781634096844344d05dbf9c0d5252ec44d8d87db/122-ng-authentication-settings/src/main/java/io/harness/ng/authenticationsettings/resources/AuthenticationSettingsResource.java#L208)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

Updates the authentication mechanism for the given Account ID.
</blockquote></details>

Done.

**Comment by boopesh007** [comment]
*[`122-ng-authentication-settings/src/main/java/io/harness/ng/authenticationsettings/resources/AuthenticationSettingsResource.java` line 229 at r3](https://reviewable.io/reviews/harness/harness-core/34074#-N3zShItvmPc_VxwrIle-r3-229:-N4C6VOH8zn6a2ShXfhO:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/781634096844344d05dbf9c0d5252ec44d8d87db/122-ng-authentication-settings/src/main/java/io/harness/ng/authenticationsettings/resources/AuthenticationSettingsResource.java#L229)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

Updates the whitelisted domains.
</blockquote></details>

Done.

**Comment by boopesh007** [comment]
*[`122-ng-authentication-settings/src/main/java/io/harness/ng/authenticationsettings/resources/AuthenticationSettingsResource.java` line 294 at r3](https://reviewable.io/reviews/harness/harness-core/34074#-N3zShItvmPc_VxwrIle-r3-294:-N4C6a4jEsH9loNb_tG7:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/781634096844344d05dbf9c0d5252ec44d8d87db/122-ng-authentication-settings/src/main/java/io/harness/ng/authenticationsettings/resources/AuthenticationSettingsResource.java#L294)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

Updates the SAML metadata for the given Account ID.
</blockquote></details>

Done.

**Comment by boopesh007** [comment]
*[`122-ng-authentication-settings/src/main/java/io/harness/ng/authenticationsettings/resources/AuthenticationSettingsResource.java` line 337 at r3](https://reviewable.io/reviews/harness/harness-core/34074#-N3zShItvmPc_VxwrIle-r3-337:-N4C6hEiDksHe4oJ_Lvx:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/781634096844344d05dbf9c0d5252ec44d8d87db/122-ng-authentication-settings/src/main/java/io/harness/ng/authenticationsettings/resources/AuthenticationSettingsResource.java#L337)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

Deletes SAML metadata for the given Account ID.
</blockquote></details>

Done.

**Comment by boopesh007** [comment]
*[`122-ng-authentication-settings/src/main/java/io/harness/ng/authenticationsettings/resources/AuthenticationSettingsResource.java` line 356 at r3](https://reviewable.io/reviews/harness/harness-core/34074#-N3zShItvmPc_VxwrIle-r3-356:-N4C6oh_59Q_QPeRrqUO:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/781634096844344d05dbf9c0d5252ec44d8d87db/122-ng-authentication-settings/src/main/java/io/harness/ng/authenticationsettings/resources/AuthenticationSettingsResource.java#L356)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

Tests SAML connectivity for the given Account ID.
</blockquote></details>

Done.

**Comment by boopesh007** [comment]
*[`122-ng-authentication-settings/src/main/java/io/harness/ng/authenticationsettings/resources/AuthenticationSettingsResource.java` line 374 at r3](https://reviewable.io/reviews/harness/harness-core/34074#-N3zShItvmPc_VxwrIle-r3-374:-N4C6v5S3IkmIHIm58RV:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/781634096844344d05dbf9c0d5252ec44d8d87db/122-ng-authentication-settings/src/main/java/io/harness/ng/authenticationsettings/resources/AuthenticationSettingsResource.java#L374)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

Sets Two-Factor authorization for the given Account ID.
</blockquote></details>

Done.

**Comment by boopesh007** [comment]
*[`400-rest/src/main/java/software/wings/resources/UserGroupResource.java` line 393 at r3](https://reviewable.io/reviews/harness/harness-core/34074#-N3zShItvmPc_VxwrIlf-r3-393:-N4C75dE9I66S8FRwpaw:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/781634096844344d05dbf9c0d5252ec44d8d87db/400-rest/src/main/java/software/wings/resources/UserGroupResource.java#L393)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

Unlinks user group from an SSO group.
</blockquote></details>

Done.

**Comment by boopesh007** [comment]
*[`400-rest/src/main/java/software/wings/resources/UserResourceNG.java` line 285 at r3](https://reviewable.io/reviews/harness/harness-core/34074#-N3zShItvmPc_VxwrIlh-r3-285:-N4C7hu42IFmuGIPd2OJ:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/781634096844344d05dbf9c0d5252ec44d8d87db/400-rest/src/main/java/software/wings/resources/UserResourceNG.java#L285)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

Changes user password.
</blockquote></details>

Done.

**Comment by boopesh007** [comment]
*[`400-rest/src/main/java/software/wings/resources/UserResourceNG.java` line 313 at r3](https://reviewable.io/reviews/harness/harness-core/34074#-N3zShItvmPc_VxwrIlh-r3-313:-N4C7sly1nsna0XRY_55:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/781634096844344d05dbf9c0d5252ec44d8d87db/400-rest/src/main/java/software/wings/resources/UserResourceNG.java#L313)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

Updates user information for the given user ID.
</blockquote></details>

Done.

**Comment by boopesh007** [comment]
*[`400-rest/src/main/java/software/wings/resources/UserGroupResource.java` line 421 at r3](https://reviewable.io/reviews/harness/harness-core/34074#-N3zShItvmPc_VxwrIlf-r3-421:-N4CGTzJ4_aM79V0Xtvl:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/781634096844344d05dbf9c0d5252ec44d8d87db/400-rest/src/main/java/software/wings/resources/UserGroupResource.java#L421)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

Links the given user group to a SAML group.
</blockquote></details>

Done.



_Code quote:_
```Java
  @Path("{userGroupId}/link/saml/{samlId}")
```

**Comment by boopesh007** [comment]
*[`400-rest/src/main/java/software/wings/resources/UserResourceNG.java` line 360 at r3](https://reviewable.io/reviews/harness/harness-core/34074#-N3zShItvmPc_VxwrIlh-r3-360:-N4CH-0PBD5QT2W_AYmM:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/781634096844344d05dbf9c0d5252ec44d8d87db/400-rest/src/main/java/software/wings/resources/UserResourceNG.java#L360)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

Creates Two-Factor authorization settings.
</blockquote></details>

Done.

**Comment by boopesh007** [comment]
*[`400-rest/src/main/java/software/wings/resources/UserResourceNG.java` line 377 at r3](https://reviewable.io/reviews/harness/harness-core/34074#-N3zShItvmPc_VxwrIlh-r3-377:-N4CHBfY57TgFALGImnA:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/781634096844344d05dbf9c0d5252ec44d8d87db/400-rest/src/main/java/software/wings/resources/UserResourceNG.java#L377)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

Enables Two-Factor authorization
</blockquote></details>

Done.

**Comment by boopesh007** [comment]
*[`400-rest/src/main/java/software/wings/resources/UserResourceNG.java` line 394 at r3](https://reviewable.io/reviews/harness/harness-core/34074#-N3zShItvmPc_VxwrIlh-r3-394:-N4CHZEw5V98quo8ERxW:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/781634096844344d05dbf9c0d5252ec44d8d87db/400-rest/src/main/java/software/wings/resources/UserResourceNG.java#L394)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

Disables Two-Factor authorization.
</blockquote></details>

Done.

**Review by rashminandasahoo — APPROVED** [review]
:lgtm:

> Reviewed 3 of 4 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34074#-:-N4CIl1X0eA_YzAIoTBf:bnfp4nl)* status: 4 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by VikasMaddukuriHarness — APPROVED** [review]
> Reviewed 4 of 4 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34074)* status: 1 file(s) to review. 1. unresolved discussions.

**Review by VikasMaddukuriHarness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34074)* status: 1 file(s) to review. 1. unresolved discussions.

**Comment by boopesh007** [comment]
*[`400-rest/src/main/java/software/wings/resources/UserGroupResource.java` line 421 at r3](https://reviewable.io/reviews/harness/harness-core/34074#-N3zShItvmPc_VxwrIlf-r3-421:-N4CLApm5iQKKL1h50Jj:bb6un1j) ([raw file](https://github.com/harness/harness-core/blob/781634096844344d05dbf9c0d5252ec44d8d87db/400-rest/src/main/java/software/wings/resources/UserGroupResource.java#L421)):*
<details><summary><i>Previously, boopesh007 (Boopesh Shanmugam) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Done

**Review by VikasMaddukuriHarness — APPROVED** [review]
> Reviewed 1 of 2 files at r3.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34074)* status: 1 file(s) to review. 1. unresolved discussions.

**Review by rashminandasahoo — APPROVED** [review]
> Reviewed 1 of 4 files at r5.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34074)* status: 1. unresolved discussions.

File: 400-rest/src/main/java/software/wings/resources/UserResourceNG.java [hunk 9]
```diff
@@ -489,8 +533,16 @@ public RestResponse<Collection<FeatureFlag>> getFeatureFlags(@PathParam("account
 
   @PUT
   @Path("/unlock-user")
-  public RestResponse<Optional<UserInfo>> unlockUser(
-      @NotEmpty @QueryParam("email") String email, @NotEmpty @QueryParam("accountId") String accountId) {
+  @ApiOperation(value = "Unlock user", nickname = "unlockUser")
+  @Operation(operationId = "unlockUser", summary = "Unlocks a locked user",
```
Comment (rashminandasahoo) [inline 400-rest/src/main/java/software/wings/resources/UserResourceNG.java:537]: Unlock a user.

**Pull request merged**